### PR TITLE
feat: use particle signature as id for actors [NET-540]

### PR DESCRIPTION
**Pull request by justprosh**
## Description
Use particle signature instead of particle id in actors ~~and particle data store~~.
UPD: use particle signature for data store will be implemented with the one of the next avm updates, because former particle id is used as salt for avm signatures

## Motivation
Anyone can set any particle id, so it is safer to use a unique particle signature for such purposes.

## Proposed Changes
- use particle signature in actors hashmap
- ~~use particle signature in hex for particle data store~~
- leave all other logs and traces with legacy particle id for now for easy debugging

## Checklist
- [ ] The code follows the project's coding conventions and style guidelines.
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: aquamarine/src/plumber.rs [hunk 1]
```diff
@@ -43,7 +43,8 @@ use crate::particle_effects::RoutingEffects;
 use crate::particle_functions::Functions;
 use crate::vm_pool::VmPool;
 
-type ParticleId = String;
+/// particle signature is used as a particle id
+type ParticleId = Vec<u8>;
```
Comment (gurinderu) [inline aquamarine/src/plumber.rs:47]: Probable it is better to use a `struct ParticleId(Vec<u8>)` here.

File: aquamarine/src/plumber.rs [hunk 2]
```diff
@@ -238,7 +241,7 @@ impl<RT: AquaRuntime, F: ParticleFunctionStatic> Plumber<RT, F> {
                 );
                 // cleanup files and dirs after particle processing (vault & prev_data)
                 // TODO: do not pass vm https://github.com/fluencelabs/fluence/issues/1216
-                if let Err(err) = actor.cleanup(particle_id, &worker_id.to_string(), &mut vm) {
+                if let Err(err) = actor.cleanup(&particle_id, &worker_id.to_string(), &mut vm) {
```
[the remark this answers is not in the document]
Comment (justprosh) [reply aquamarine/src/plumber.rs:244]: DataStore trait from avm libs accepts particle_id as &str for file names

[the remark this answers is not in the document]
Comment (justprosh) [reply aquamarine/src/plumber.rs:244]: hm, you are right, i can do a conversion to hex inside actor.cleanup

**Review by folex — APPROVED** [review]

File: aquamarine/src/particle_executor.rs [hunk 3]
```diff
@@ -81,7 +81,7 @@ impl<RT: AquaRuntime> ParticleExecutor for RT {
                         current_peer_id: Cow::Owned(current_peer_id.to_string()),
                         init_peer_id: Cow::Owned(particle.init_peer_id.to_string()),
                         // we use signature hex as particle id to prevent compromising of particle data store
-                        particle_id: Cow::Owned(particle.get_signature_hex()),
+                        particle_id: Cow::Borrowed(&particle.id),
```
Comment (gurinderu) [inline aquamarine/src/particle_executor.rs:84]: Interesting, why do we need a copy-on-write type here? I assume we only read them

File: aquamarine/src/actor.rs [hunk 4]
```diff
@@ -99,15 +99,21 @@ where
...
+    pub fn cleanup(&self, vm: &mut RT) {
+        tracing::debug!(
+            target: "particle_reap",
+            particle_id = self.particle.id, worker_id = self.current_peer_id.to_string(),
+            "Reaping particle's actor"
+        );
         // TODO: remove dirs without using vm https://github.com/fluencelabs/fluence/issues/1216
-        vm.cleanup(particle_id, current_peer_id)?;
-        Ok(())
+        // TODO: ??? we don't have this issue anymore
```
Comment (gurinderu) [inline aquamarine/src/actor.rs:109]: We still  need the vm instance to remove directories,  but doesn't have issue.

**Review by gurinderu — APPROVED** [review]

File: aquamarine/src/particle_executor.rs (hunk 3 above)
Comment (justprosh) [reply aquamarine/src/particle_executor.rs:84]: idk, ParticleParameters defined in avm

**Pull request merged**